io/include/pcl/io/openni_camera/openni_depth_image.h

```diff
@@ -163,7 +163,7 @@ namespace openni_wrapper
   } ;
 
   DepthImage::DepthImage (boost::shared_ptr<xn::DepthMetaData> depth_meta_data, float baseline, float focal_length, XnUInt64 shadow_value, XnUInt64 no_sample_value) throw ()
-  : depth_md_ (depth_meta_data)
+  : depth_md_ (std::move(depth_meta_data))
   , baseline_ (baseline)
```

Comment:
All the remaining parameters in this constructor all eligible for `std::move`.

Reply:
Nop, the are primtive data types, so `move` does nothing. This would cause another [clang-tidy warning](https://clang.llvm.org/extra/clang-tidy/checks/performance-move-const-arg.html).